feat(app,api-client,react-api-client): module calibration wizard scaffolding

b-cooper opened the pull request:
# Overview

WIP module cal happy path wizard buildout - initial work adding the scaffolding for the module calibration wizard. updates module api types and adds shared data module calibration types.

# Test Plan


# Changelog


# Review requests


# Risk assessment

smb2268 committed:
Adding calibrate module command, happy path and typing work for wizard flow

b-cooper committed:
standardize naming

b-cooper committed:
flesh out a bit

brenthagen committed:
Merge branch 'edge' into app_module-cal-flow

brenthagen committed:
add module calibration feature flag

brenthagen committed:
Merge branch 'edge' into app_module-cal-flow

brenthagen committed:
formatting

brenthagen committed:
update module wizard screens styling and text

b-cooper commented on app/src/organisms/ModuleCard/ModuleOverflowMenu.tsx:
```diff
@@ -70,9 +78,9 @@ export const ModuleOverflowMenu = (
-        <MenuItem onClick={() => handleCalibrateClick()}>
-          Calibrate
-        </MenuItem>
+        {enableModuleCalibration ? (
```
Before getting this in, can we also put a condition here so that this will only ever occur for modules on Flexes? Even after we remove the FF, we'll want that to be true

b-cooper commented on app/src/organisms/ModuleWizardFlows/index.tsx:
```diff
@@ -0,0 +1,186 @@
+        setMaintenanceRunId(response.data.id)
+      },
+    },
+    host
```
why are we passing a host override here? This should always be launched from with a ApiProvider, right?

brenthagen replied:
yes, removed

koji commented on app/src/organisms/ModuleWizardFlows/AttachProbe.tsx:
```diff
@@ -0,0 +1,71 @@
+  ${TYPOGRAPHY.pRegular};
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
```
This might be 1.25rem since we don't have 1.275rem in constants?

koji commented on app/src/organisms/ModuleWizardFlows/AttachProbe.tsx:
```diff
@@ -0,0 +1,71 @@
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
+    line-height: 1.75rem;
```
```suggestion
    line-height: ${TYPOGRAPHY.lineHeight28};
```

koji commented on app/src/organisms/ModuleWizardFlows/BeforeBeginning.tsx:
```diff
@@ -0,0 +1,64 @@
+import { StyledText } from '../../atoms/text'
+import { GenericWizardTile } from '../../molecules/GenericWizardTile'
+import { WizardRequiredEquipmentList } from '../../molecules/WizardRequiredEquipmentList'
+import type {
+  CreateMaintenanceRunData,
+  MaintenanceRun,
+} from '@opentrons/api-client'
+import type { AxiosError } from 'axios'
```
```suggestion
import type { AxiosError } from 'axios'
import type {
  CreateMaintenanceRunData,
  MaintenanceRun,
} from '@opentrons/api-client'
```

koji commented on app/src/organisms/ModuleWizardFlows/FirmwareUpdate.tsx:
```diff
@@ -0,0 +1,43 @@
+  ${TYPOGRAPHY.pRegular};
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
```
same as the above

koji commented on app/src/organisms/ModuleWizardFlows/FirmwareUpdate.tsx:
```diff
@@ -0,0 +1,43 @@
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
+    line-height: 1.75rem;
```
```suggestion
    line-height: ${TYPOGRAPHY.lineHeight28};
```

koji commented on app/src/organisms/ModuleWizardFlows/PlaceAdapter.tsx:
```diff
@@ -0,0 +1,47 @@
+  ${TYPOGRAPHY.pRegular};
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
```
same as the above

koji commented on app/src/organisms/ModuleWizardFlows/PlaceAdapter.tsx:
```diff
@@ -0,0 +1,47 @@
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
+    line-height: 1.75rem;
```
```suggestion
    line-height: ${TYPOGRAPHY.lineHeight28};
```

koji commented on app/src/organisms/ModuleWizardFlows/AttachProbe.tsx:
```diff
@@ -0,0 +1,71 @@
+
+import type { ModuleCalibrationWizardStepProps } from './types'
+
+export const BODY_STYLE = css`
```
Maybe `BODY_TEXT_STYLE` since this style const handles font style only.

koji commented on app/src/organisms/ModuleWizardFlows/SelectLocation.tsx:
```diff
@@ -0,0 +1,58 @@
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
+    line-height: 1.75rem;
```
```suggestion
    line-height: ${TYPOGRAPHY.lineHeight28};
```

koji commented on app/src/organisms/ModuleWizardFlows/SelectLocation.tsx:
```diff
@@ -0,0 +1,58 @@
+  }
+  const bodyText = (
+    <>
+      <Banner type="warning" size={SIZE_1} marginY={SPACING.spacing4}>
```
slack (from live dqa)
https://opentrons.slack.com/archives/CSCLVUW3C/p1685644862441539

```suggestion
      <Banner type="warning" size="1rem" marginY={SPACING.spacing4}>
```

koji commented on app/src/organisms/ModuleWizardFlows/Success.tsx:
```diff
@@ -0,0 +1,47 @@
+
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    font-size: 1.275rem;
+    line-height: 1.75rem;
```
```suggestion
    line-height: ${TYPOGRAPHY.lineHeight28};
```

koji commented on app/src/organisms/ModuleWizardFlows/index.tsx:
```diff
@@ -0,0 +1,186 @@
+import { Success } from './Success'
+
+import type { AttachedModule } from '@opentrons/api-client'
+import { FirmwareUpdate } from './FirmwareUpdate'
```
Can we move this line up?
Maybe after BeforeBeginning would be good.

koji commented on app/src/organisms/ModuleWizardFlows/index.tsx:
```diff
@@ -0,0 +1,186 @@
+  const host = useHost()
+  const [maintenanceRunId, setMaintenanceRunId] = React.useState<string>('')
+  const [currentStepIndex, setCurrentStepIndex] = React.useState<number>(0)
+  const totalStepCount = moduleCalibrationSteps.length - 1
```
From the design the steps counting start from `BeforeBeginning`. I guess this won't be able to count all steps properly.

koji commented on app/src/organisms/ModuleWizardFlows/index.tsx:
```diff
@@ -0,0 +1,186 @@
+  return (
+    <Portal level="top">
+      {isOnDevice ? (
+        <LegacyModalShell>
```
Really need to use `LegacyModalShell` for touchscreen app?

koji commented on app/src/organisms/ModuleWizardFlows/index.tsx:
Can we add the basic tests for this component or create a ticket to follow later?
The coverage is very low when compare to other components in this PR.

brenthagen replied:
this is an initial scaffolding commit to unblock further work - regarding but not closing RAUT-549. testing and polish work will happen in future PRs

brenthagen committed:
Merge branch 'edge' into app_module-cal-flow

brenthagen committed:
feat(app,api-client,react-api-client): module calibration wizard scaffolding

adds the scaffolding to implement the module calibration wizard. updates module api types with
moduleOffset and meta fields.

re RAUT-549

brenthagen committed:
Merge branch 'edge' into app_module-cal-flow
